mhs/common/mhs_common/workflow/tests/test_asynchronous_express.py

```diff
@@ -0,0 +1,153 @@
+import asyncio
+import unittest
+from unittest import mock
+
+import requests
+from utilities import test_utilities
+from utilities.test_utilities import async_test
+
+import mhs_common.workflow.asynchronous_express as async_express
+from mhs_common.messages import ebxml_request_envelope, ebxml_envelope
+from mhs_common.state import work_description
+from mhs_common.state.work_description import MessageStatus
+
+PARTY_KEY = 'party-key'
+MESSAGE_ID = 'message-id'
+CORRELATION_ID = 'correlation-id'
+INTERACTION_DETAILS = {'workflow': 'async-express'}
+PAYLOAD = 'payload'
+SERIALIZED_MESSAGE = 'serialized-message'
+
+
+class TestAsynchronousExpressWorkflow(unittest.TestCase):
+    def setUp(self):
+        self.mock_persistence_store = mock.MagicMock()
+        self.mock_transmission_adaptor = mock.MagicMock()
+
+        patcher = mock.patch.object(work_description, 'create_new_work_description')
+        self.mock_create_new_work_description = patcher.start()
+        self.addCleanup(patcher.stop)
+
+        patcher = mock.patch.object(ebxml_request_envelope, 'EbxmlRequestEnvelope')
+        self.mock_ebxml_request_envelope = patcher.start()
+        self.addCleanup(patcher.stop)
+
+        self.workflow = async_express.AsynchronousExpressWorkflow(self.mock_persistence_store,
+                                                                  self.mock_transmission_adaptor, PARTY_KEY)
+
+    def tearDown(self):
+        self.mock_create_new_work_description.assert_called_once_with(self.mock_persistence_store, MESSAGE_ID,
+                                                                      MessageStatus.OUTBOUND_MESSAGE_RECEIVED)
+        self.mock_ebxml_request_envelope.assert_called_once()
```

Comment:
Feels a bit odd using the tear down method for assertions, and if a test is ever added that doesn't  perform all the same actions the current tests do this tear down might not apply.